shared/src/main/scala/org/jetbrains/sbt/structure/data.scala

```diff
@@ -126,15 +129,22 @@ case class ScalaData(
   compilerJars: Seq[File],
   extraJars: Seq[File],
   compilerBridgeBinaryJar: Option[File],
-  options: Seq[String]
+  options: Map[Configuration, Seq[String]]
 ) {
   def allJars: Seq[File] = libraryJars ++ compilerJars ++ extraJars
   def allCompilerJars: Seq[File] = libraryJars ++ compilerJars
 }
 
-case class DependencyData(projects: Seq[ProjectDependencyData],
-                          modules: Seq[ModuleDependencyData],
-                          jars: Seq[JarDependencyData])
+case class DependencyData(projects: Dependencies[ProjectDependencyData],
+                          modules: Dependencies[ModuleDependencyData],
+                          jars: Dependencies[JarDependencyData])
+
+/**
+ * When the project is imported without prod/test sources feature enabled, all dependencies are put in forProduction parameter.
+ * @param forTest dependencies that should go to the test module
+ * @param forProduction dependencies that should go to the main module
+ */
+case class Dependencies[T](forTest: Seq[T], forProduction: Seq[T])
```

Comment:
(minor, but still)
A more natural, intuitive way is to put prod first and test second as production classes are considered "main"

Reply:
Same for fields in `org.jetbrains.sbt.structure.ProjectData`